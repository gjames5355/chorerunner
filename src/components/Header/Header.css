/* 
font-family: 'Press Start 2P', cursive;
font-family: 'Fredoka One', cursive;
font-family: 'Concert One', cursive;
font-family: 'Electrolize', sans-serif; */

header {
  height: 55px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  width: 100%;
  background-color: rgba(86, 74, 216, 1);
  z-index: 1;
}

header>h1 {
  line-height: 11px;
  width: 40%;
}

h1>a {
  color: #FFD900;
  font-size: 15px;
  padding-left: 10px;
  font-family: 'Press Start 2P', cursive;
}

nav a {
  color: #FFD900;
  font-weight: 600px;
  padding: 0 20px;
  font-size: 18px;
  font-family: 'Electrolize', sans-serif;
  /* display:block; */
}

nav {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.loggedin-info {
  display: flex;
}

.loggedin-info span {
  color: black;
  padding-right: 20px;
  display: none;
}

@media (min-width: 475px) {
  header>h1 {
    line-height: 11px;
    width: 200px;
  }

  h1>a {
    color: #FFD900;
    font-size: 15px;
    padding-left: 10px;
    font-family: 'Press Start 2P', cursive;
  }
}