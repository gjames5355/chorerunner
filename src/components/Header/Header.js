import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import UserContext from '../../contexts/UserContext';
import './Header.css';

class Header extends Component {
  static contextType = UserContext;

  handleLogoutClick = () => {
    this.context.processLogout();
  };

  renderLogoutLink() {
    return (
      <div className='loggedin-info'>
        <span>Hi Owner!</span>
        <nav>
          <Link
            style={{ textDecoration: 'none' }}
            onClick={this.handleLogoutClick}
            to='/'
          >
            Logout
          </Link>
        </nav>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <nav>
        <Link style={{ textDecoration: 'none' }} to='/login'>
          Login
        </Link>{' '}
        <Link style={{ textDecoration: 'none' }} to='/register'>
          Sign up
        </Link>
      </nav>
    );
  }

  render() {
    let display = (
      <h1>
        {TokenService.hasAuthToken() ? (
          <Link
            to={
              this.context.user.type === 'user'
                ? '/parent-dashboard'
                : '/member-dashboard'
            }
            style={{ textDecoration: 'none' }}
          >
            Chore Runner
          </Link>
        ) : (
          <Link to='/' style={{ textDecoration: 'none' }}>
            Chore Runner
          </Link>
        )}
      </h1>
    );

    return (
      <>
        {display}
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </>
    );
  }
}

export default Header;
