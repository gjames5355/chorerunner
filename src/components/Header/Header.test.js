import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
