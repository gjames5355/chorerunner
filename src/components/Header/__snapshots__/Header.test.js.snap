// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component testing renders UI as expected 1`] = `
Array [
  <h1>
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Chore Runner
    </a>
  </h1>,
  <nav>
    <a
      href="/login"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Login
    </a>
     
    <a
      href="/register"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Sign up
    </a>
  </nav>,
]
`;

exports[`Header component testing renders empty given no tabs 1`] = `
<Fragment>
  <h1>
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/"
    >
      Chore Runner
    </Link>
  </h1>
  <nav>
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/login"
    >
      Login
    </Link>
     
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/register"
    >
      Sign up
    </Link>
  </nav>
</Fragment>
`;
